src/ameba/rule/style/is_a_filter.cr

```diff
@@ -77,8 +80,16 @@ module Ameba::Rule::Style
         end
       end
 
-      issue_for node.name_location, end_location,
-        MSG % {node.name, name, node.name}
+      old = OLD % node.name
+      new = NEW % {node.name, name}
+      msg = MSG % {new, old}
+      if end_location
+        issue_for(filter_location, end_location, msg) do |corrector|
+          corrector.replace(filter_location, end_location, new)
+        end
```

Comment:
can this two conditions be merged into a single one? 

```crystal
issue_for(filter_location, end_location, msg) do |corrector|
  corrector.replace(filter_location, end_location, new) if end_location
end
````

Reply:
No, not currently because the presence of a block is used to infer whether an issue can be autocorrected. RuboCop works around this by creating a new `Corrector` instance for every block and then checking if the corrector is empty.